std/experimental/ndslice/selection.d

```diff
@@ -0,0 +1,1386 @@
+/**
+$(SCRIPT inhibitQuickIndex = 1;)
+
+Селекторы создают новый тип итератора для тех же данных.
+Сами данные остаются неизменными (to translate).
+
+$(H2 Subspace selectors)
+
+The destination of subspace selectors is painless generalization and combination of other selectors.
+`pack!K` creates a slice of slices `Slice!(N-K, Slice!(K+1, Range))` by packing last `K` dimensions of highest pack of dimensions,
+so type of element of `slice.byElement` is `Slice!(K, Range)`.
+Another way to use `pack` is transposition of packs of dimensions using `evertPack`.
+Examples with subspace selectors are available for selectors, $(SUBREF slice, Slice.shape), $(SUBREF slice, .Slice.elementsCount).
+
+$(BOOKTABLE ,
+
+$(TR $(TH Function Name) $(TH Description))
+$(T2 pack, returns slice of slices.)
+$(T2 unpack, unites all dimension packs.)
+$(T2 evertPack, reverse packs of dimensions.)
+)
+
+$(BOOKTABLE $(H2 Selectors),
+
+$(TR $(TH Function Name) $(TH Description))
+$(T2 blocks, n-dimensional slice of n-dimensional non-overlapping blocks)
+$(T2 windows, n-dimensional slice of n-dimensional overlapping  windows)
+$(T2 diagonal, 1-dimensional slice of diagonal elements)
+$(T2 reshape, returns new slice for the same data)
+$(T2 byElement, a random access range of all elements)
+$(T2 byElementInStandardSimplex, an input range of standard simplex in hypercube (left upper triangular matrix).)
+$(T2 indexSlice, returns a slice with elements equals to initial index.)
+)
+
+License:   $(WEB www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+
+Authors:   Ilya Yaroshenko
+
+Source:    $(PHOBOSSRC std/_experimental/_ndslice/_selection.d)
+
+Macros:
+SUBMODULE = $(LINK2 std_experimental_ndslice_$1.html, std.experimental.ndslice.$1)
+SUBREF = $(LINK2 std_experimental_ndslice_$1.html#.$2, $(TT $2))$(NBSP)
+T2=$(TR $(TDNW $(LREF $1)) $(TD $+))
+T4=$(TR $(TDNW $(LREF $1)) $(TD $2) $(TD $3) $(TD $4))
+*/
+module std.experimental.ndslice.selection;
+
+import std.traits;
+import std.meta; //: allSatisfy;
+
+import std.experimental.ndslice.internal;
+import std.experimental.ndslice.slice; //: Slice;
+
+/++
+Packs a slice into the composed slice, i.e. slice of slices.
+Params:
+    K = sizes of packs of dimensions
+Returns:
+    `pack!K` returns `Slice!(N-K, Slice!(K+1, Range))`;
```

Comment:
confusing because N and Range aren't in `pack!K`. More complete expressions would be better.